### ISSUE TITLE
Redirect being followed with DELETE rather than GET intermittently

### ISSUE THREAD

**Issue by christoomey** [inertiajs/inertia#419]
### Versions:

- `@inertiajs/inertia` version: 0.8.2
- `@inertiajs/inertia-svelte` version: 0.6.0

### Describe the problem:

I'm seeing an intermittent issue where after issuing a DELETE request which returns a 302, the subsequent request (following the redirect to the provided "Location") is also made with a DELETE, rather than a GET as expected.

### Steps to reproduce:

I'm specifically running into this with the "Sign Out" button in my app.

- User clicks "Sign Out" button
- WE fire an Inertia DELETE to /sign_out
- The delete is processed, the user is signed out, and the response comes back as a 302 w/ location /sign_in
- Instead of issuing a `GET` to /sign_in, a DELETE request is issued. This ends up not matching any route, so a 404 error is displayed in the error modal

I'm only able to reproduce sporadically and cannot determine any other timing or sequencing unfortunately. I tried digging into the code but wasn't able to determine exactly where the "follow the redirect" logic is, but wondering if there might be a race condition or caching of the request "method" issue at play?

**Comment by RobertBoes** [comment]
The server-side adapter should return a 303 response, instead of 302, see https://inertiajs.com/redirects#303-response-code

A 302 redirect means the browser should follow current protocol, so if the request was made using DELETE it should follow op the redirect also with a DELETE request. Inertia changes this for the official adapters to 303 so the follow-up request can be made as a GET request ([inertia-laravel src](https://github.com/inertiajs/inertia-laravel/blob/master/src/Middleware.php#L121-L131), [inertia-rails src](https://github.com/inertiajs/inertia-rails/blob/8c62d3aec7b5fb05544307b4bdf33cfc2b61aa1e/lib/inertia_rails/middleware.rb#L18)). Which server-side adapter are you using?

**Comment by reinink** [comment]
Yup, exactly @RobertBoes. 303 redirects are key to making Inertia apps work.

That said, I'm pretty sure the Rails adapter (which @christoomey uses) does automatically set this:

https://github.com/inertiajs/inertia-rails/blob/8c62d3aec7b5fb05544307b4bdf33cfc2b61aa1e/lib/inertia_rails/middleware.rb#L18

Maybe there is a bug with that implementation.

**Comment by christoomey** [comment]
Just when I thought I couldn't love this project any more :) TIL 302 vs 303 distinction, and I just love how solidly this project builds on the foundations of the web. :chefskiss:

Based on your replies I've now been able to verify that while the vast majority of requests do return a 303, but the error case outlined above occurs when the status code is 302. I'm not sure _why_ this small percentage of responses have the wrong status code, but it's clear that Inertia on the client side is doing exactly as it is supposed to.

Closing this issue now. I'll do a bit more digging to see if this is an issue in my apps code, or possibly something to review with the inertia-rails repo.

Thanks!

**Comment by reinink** [comment]
@christoomey Haha, awesome buddy! I had the exact same experience learning about 302 vs 303 redirects. This was early on when I was still trying to figure out if Inertia was a legitimate approach to building web apps. When I ran into this, I was like "oh crap, I've finally found the thing that's going to prove this approach wrong". And then I learned about the 303 redirects, and was like 🤯😍🙌.

I actually had a similar experience discovering the `Vary: Accept` response header...also super critical to making Inertia apps work (see [here](https://github.com/inertiajs/inertia-laravel/blob/6d4276984ccc58067056708cd3f2684fdf2ab095/src/Response.php#L95) and [here](https://github.com/inertiajs/inertia-rails/blob/c19bf06e93dee8bbb0f105e287362656b00eff72/lib/inertia_rails/renderer.rb#L19)). This tells the browser that an Inertia response can either be HTML (the initial page visit), or JSON (the Inertia visit). Inertia uses these core fundamental HTTP tools/principles to make the whole system work.

Anyway, if you figure out why you're not getting a 303 in your apps, be sure to let us know. I'm curious where this is breaking down for you.
